external/model-preparation-algorithm/mpa_tasks/apis/classification/task.py

```diff
@@ -104,19 +93,51 @@ def infer(self,
 
         dataset_size = len(dataset)
         for i, (dataset_item, prediction_item) in enumerate(zip(dataset, predictions)):
-            label = []
+            item_labels = []
+            pos_thr = 0.5
+
             if self._multilabel:
-                pos_thr = 0.5
                 for cls_idx, pred_item in enumerate(prediction_item):
                     if pred_item > pos_thr:
                         cls_label = ScoredLabel(self.labels[cls_idx], probability=float(pred_item))
-                        label.append(cls_label)
+                        item_labels.append(cls_label)
+                    else:
+                        logger.info('item_labels is empty.')
+                        item_labels.append(ScoredLabel(self._empty_label, probability=1.))
+
+            elif self._hierarchical:
+                for i in range(self._hierarchical_info['num_multiclass_heads']):
```

Comment:
Can't we take out the codes in this part as a function? (If possible?)

Reply:
Surely, I prefer to make a function to parse hierarchical information. I'll think about this carefully.

Reply:
Hi, hierarchical information is indexing on infer and mpa_cls_dataset.py. But the actual work is a little different. Of course, there will be nice ways to parse the information, but the way should cover HierarchicalClsHead in MPA. So, I'll leave the code as-is and I'll refactor later.

Reply:
Is there any problem that 'i' overlaps with the index of dataset_item? 
Is there a reason why you have to use it as an "i"?

And it's still hard to understand why it's hard to separate it into functions.

Reply:
Okay, I'll clear index overlapping issues. The hierarchical information indexing looks like "Similar" with the thing in MPAClsDataset, but it actually works differently.

Reply:
I didn't mean to combine the same function as a function, but I wanted to have a separate referral function for multi-label or h-label.